app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java

```diff
@@ -0,0 +1,168 @@
+                        if (StringUtils.isEmpty(input)) {
+                            Snackbar.make(Objects.requireNonNull(getActivity()).findViewById(android.R.id.content),
+                                    R.string.snackbar_name_not_blank,
+                                    Snackbar.LENGTH_LONG).show();
+                        } else {
+                            String mFilename = input.toString();
+                            try {
+                                createPdf(TEXT_FILE_URI, mFilename);
+                                Snackbar.make(Objects.requireNonNull(getActivity()).findViewById(android.R.id.content),
+                                        R.string.snackbar_pdfCreated,
+                                        Snackbar.LENGTH_LONG).show();
+                            } catch (DocumentException | IOException e) {
+                                e.printStackTrace();
+                            }
+                        }
+                    }
+                })
+                .show();
+    }
+    public void createPdf(Uri fileURI, String outputFile)
+            throws DocumentException, IOException {
+
+        Document document = new Document();
+        String finalOutput = Environment.getExternalStorageDirectory() + "/" + "PDFfiles" + "/" + outputFile + ".pdf";
+        PdfWriter.getInstance(document, new FileOutputStream(finalOutput)).setPdfVersion(PdfWriter.VERSION_1_7);
+
+        document.open();
+        Font myfont = new Font();
+        myfont.setStyle(Font.NORMAL);
+        myfont.setSize(11);
+
+        document.add(new Paragraph("\n"));
+        readTextFile(fileURI, document, myfont);
+        document.close();
+    }
+    private void readTextFile(Uri uri, Document document, Font myfont) {
+        InputStream inputStream;
+        try {
+            inputStream = getActivity().getContentResolver().openInputStream(uri);
+            BufferedReader reader = new BufferedReader(new InputStreamReader(
+                    inputStream));
+
+            String line;
+            while ((line = reader.readLine()) != null) {
+                Paragraph para = new Paragraph(line + "\n", myfont);
+                para.setAlignment(Element.ALIGN_JUSTIFIED);
+                document.add(para);
+            }
+            reader.close();
+            inputStream.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+    private void textFileSelect() {
+
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.setType("*/*");
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+
+        try {
+            startActivityForResult(
+                    Intent.createChooser(intent, "Select a File to Upload"),
+                    FILE_SELECT_CODE);
+        } catch (android.content.ActivityNotFoundException ex) {
+            Snackbar.make(Objects.requireNonNull(getActivity()).findViewById(android.R.id.content),
```

Comment:
Don't use getActivity() function. Create a new private variable `private Activity mActivity;` and assign it value in `onAttach` (just like other fragments. See MergeFilesFragment)